manager/middlewares/authentication.go

```diff
@@ -22,12 +30,24 @@ func Authenticator() gin.HandlerFunc {
 			c.AbortWithStatusJSON(http.StatusUnauthorized, utils.ErrorResponse{Error: "Invalid x-rh-identity header"})
 			return
 		}
-		c.Set(KeyAccount, identity.Identity.AccountNumber)
+		AccountIDCache.Lock.Lock()
```

Comment:
Is it needed to lock it for both read and write? Isn't locking for write enought?

Reply:
Not sure, anyway, one mutex for short time in this piece of code will not affect performance.